cmd/kops/edit_cluster.go

```diff
@@ -155,5 +157,29 @@ func RunEditCluster(f *util.Factory, cmd *cobra.Command, args []string, out io.W
 		return fmt.Errorf("error writing completed cluster spec: %v", err)
 	}
 
+	// Convert the cluster back to YAML for comparison purposes
+	newYaml, err := api.ToVersionedYaml(newCluster)
+	if err != nil {
+		return err
+	}
+
+	// Marshal the edited YAML to a map; this will prevent bad diffs due to sorting
+	var editedYamlObj map[string]interface{}
+	err = utils.YamlUnmarshal([]byte(edited), &editedYamlObj)
+	if err != nil {
+		return err
+	}
+
+	// Convert the object back to YAML so that we can compare it to the cluster YAML
+	editedYaml, err := utils.YamlMarshal(editedYamlObj)
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Equal(editedYaml, newYaml) {
+		discardedChanges := diff.FormatDiff(string(editedYaml), string(newYaml))
+		fmt.Fprintln(os.Stderr, "Not all edits were applied:\n"+discardedChanges)
```

Comment:
@chrislovecnm @sethpollack They are related but non-overlapping. #1741 does not address silent YAML parse field dropping.